### ISSUE TITLE
How to impl zero-copy  when send a message out-of-process msgs to cuda ? 

### ISSUE THREAD

**Issue by lix19937** [NVIDIA-ISAAC-ROS/isaac_ros_nitros#45]

**Comment by ZhenshengLee** [comment]
> I found, if user want to leverage the benefit of zero-copy in NITROS, all NITROS-accelerated nodes must run in the same process.

Yes, the root cause of the performance improvement from isaac_ros_nitros is TYPE ADAPTATION https://ros.org/reps/rep-2007.html and TYPE NEGOTIATION https://ros.org/reps/rep-2009.html .

These two feature needs intra-communication.

more info:
https://www.openrobotics.org/blog/2022/5/24/ros-2-humble-hawksbill-release
https://developer.nvidia.com/blog/improve-perception-performance-for-ros-2-applications-with-nvidia-isaac-transport-for-ros/

**Comment by ZhenshengLee** [comment]
If you don't use the ros2 transport, you could use nvsci/cuda_ipc to get cuda sharedmemory.

more info:
https://github.com/pytorch/pytorch/issues/137680

**Comment by ZhenshengLee** [comment]
@lix19937 Do you work in China? You could contact me with wechat to discuss:   zhensheng_li
